agent/rpm/Makefile

```diff
@@ -1,26 +1,54 @@
-# Makefile for generating a source RPM and optional local RPM
+# Makefile for generating a source RPM and, optionally, binary RPMs
 # for the Pbench agent.
 
-# To limit the builds to certain chroots or exclude certain chroots
-# from building, add entries of the form
+# When building a binary RPM on COPR, to limit the builds to certain chroots
+# or to exclude certain chroots from building, add entries of the form
 #    "--chroot centos-stream-9-x86_64"
 # or
 #    "--exclude-chroot centos-stream-9-x86_64"
-# to the CHROOTS variable below.
+# to the definition of the CHROOTS variable.
+#
 # Multiple such entries can be added to be passed as options to
-# `copr-cli build'.  By default, we build every chroot configured for
-# the repo.
-# N.B. `copr-cli' flags an error if the value of a `--chroot' or
-# `--exclude-chroot' option is not configured in the repo.
+# `copr-cli build'.  N.B. `copr-cli' flags an error if the value of a
+# `--chroot' or `--exclude-chroot' option is not configured in the repo.
 # E.g. to build the RHEL9 chroots only:
 # CHROOTS = --chroot centos-stream-9-x86_64 \
 #           --chroot centos-stream-9-aarch64 \
 #           --chroot epel-9-x86_64 \
 #           --chroot epel-9-aarch64
-# By default, we build every chroot enabled in the repo.
+#
+# Unless overridden on the command line, build every chroot enabled in the repo.
 CHROOTS =
 
+# These targets build the binary RPM for the specified distro version; the "ci"
+# target builds them all.
+RHEL_RPMS = rhel-9-rpm rhel-8-rpm rhel-7-rpm
+CENTOS_RPMS = centos-9-rpm centos-8-rpm
+FEDORA_RPMS = fedora-34-rpm fedora-35-rpm fedora-36-rpm
+ALL_RPMS = ${RHEL_RPMS} ${CENTOS_RPMS} ${FEDORA_RPMS}
+
 component = agent
 subcomps = agent
 
-include ../../utils/rpm.mk
+export  # Export the definitions above in the invocation of the sub-make.
+
+# For historical compatibility, this is the default target.
+all: srpm
+
+# For the CI, we build all of our favorite RPMs, but we do that via the
+# dependency list rather than by any immediate action.  There are two things
+# happening here:  first, the dependencies can be built in parallel if the
+# invocation permits it, e.g., 'make -j 4 -O ci'; second, rpm.mk, when it is
+# invoked, determines which distro to build for based on the pattern of the
+# specified target, so having multiple targets in the same invocation for
+# different distributions won't work, so we need to invoke it separately for
```

Comment:
Really dislike the "chained ', so'" clauses here. In fact replacing the first ", so having" with ". Having" would probably make this long paragraph flow a bit more smoothly.

Reply:
OK, but I don't really want to break the second reason into multiple sentences when both it and the first reason are introduced in the same sentence.  I rejiggered things a bit in my update -- see if what I've now got will suffice -- otherwise, I'll break it into multiple sentences.

Reply:
It's still an awkward long sentence, but as I was objecting more to the repetition I won't object any further. 😺